### ISSUE TITLE
Potential Data Contamination in Pretraining Phase Affecting Zero-Shot Evaluation

### ISSUE THREAD

**Issue by mirmuss** [moment-timeseries-foundation-model/moment-research#17]
Hi there! 👋
Thank you for this interesting work on time series foundation models. While reviewing the code, I noticed an important concern regarding data exposure during pretraining that might affect the validity of downstream evaluations.
In get_timeseries_dataloader.py, when dataset_names is set to "all" during pretraining:

``` python
if args.dataset_names == "all":
    assert (
        args.task_name == "pre-training"
    ), "Only pre-training task supports all datasets"
    args.dataset_names = all_datasets
```

And looking at get_all_datasets(), it includes all available datasets:

``` python
def get_all_datasets(datasets_fraction: float = 1.0):
    classification_datasets = get_classification_datasets(collection="UCR")
    forecasting_datasets_long = get_forecasting_datasets(collection="autoformer")
    forecasting_datasets_short = get_forecasting_datasets(collection="monash")
    forecasting_datasets_short = forecasting_datasets_short + get_forecasting_datasets(
        collection="fred/preprocessed"
    )
    anomaly_detection_datasets = get_anomaly_detection_datasets(
        collection="TSB-UAD-Public"
    )
```

This means that during pretraining, the model is exposed to ALL datasets, including those that might be used later for evaluating zero-shot or few-shot capabilities. This exposure creates a significant data contamination issue:

1. Any zero-shot learning evaluation would not be truly "zero-shot" as the model has already seen the data during pretraining
2. Performance metrics on downstream tasks might be artificially inflated due to this data leakage
3. The true generalization capability of the model to unseen datasets cannot be accurately assessed

Suggested Solution:
Consider maintaining a strict separation of datasets by:

1. Creating a held-out test set of datasets that are completely unused during pretraining
2. Documenting which datasets were used in pretraining to ensure transparent evaluation
3. Implementing a mechanism to explicitly exclude certain datasets from the pretraining phase

Would love to hear your thoughts on this!

**Comment by mononitogoswami** [comment]
Hi, 
Thank you for your interest in MOMENT. This is indeed a good question. 

The time series used for pre-training and evaluation are disjoint, and therefore there is no contamination per se. 

For forecasting datasets, e.g. ETTh1, the training and validation time series are used for pre-training. The model doesn't see the testing split. Similarly, for classification and anomaly detection, the model is only trained on training time series.

> Any zero-shot learning evaluation would not be truly "zero-shot" as the model has already seen the data during pre-training

I agree with you in principle. However, we define zero-shot as _zero parameter updates_ on the testing dataset. 

> Performance metrics on downstream tasks might be artificially inflated due to this data leakage

I think performance is relative. The fact that MOMENT pre-trained on the same training data as the baselines, without task-specific inductive biases, can outperform them in some settings, is promising. I disagree that there is any data leakage, given the training and testing sets are disjoint.

> The true generalization capability of the model to unseen datasets cannot be accurately assessed

Agreed! There has been a lot of work recently by the community that has shown that MOMENT indeed generalizes to unseen datasets and domains [1, 2, 3, 4]. I believe that studies likes this are very important to ascertain the effectiveness of any foundation model, including MOMENT.

Thank you for your suggestions! We really appreciate them!

Regarding, 
> Documenting which datasets were used in pretraining to ensure transparent evaluation

While we will try to make it clearer in future iterations of MOMENT, in the current paper, we do explain the pre-training and testing splits.

## References
1. Pillai, Arvind, et al. "PaPaGei: Open Foundation Models for Optical Physiological Signals." arXiv preprint arXiv:2410.20542 (2024).
2. Yuan, Zhizhang, et al. "BrainWave: A Brain Signal Foundation Model for Clinical Applications." arXiv preprint arXiv:2402.10251 (2024).
3. Mulayim, Ozan Baris, et al. "Are Time Series Foundation Models Ready to Revolutionize Predictive Building Analytics?." Proceedings of the 11th ACM International Conference on Systems for Energy-Efficient Buildings, Cities, and Transportation. 2024.
4. Dumitru, Corneliu Octavian, et al. "Building Anomaly Detection with Self-supervised Learning. Case study: the city of Bucharest, Romania." (2024).

**Comment by mirmuss** [comment]
Thanks for your reply, 

> I agree with you in principle. However, we define zero-shot as zero parameter updates on the testing dataset.

I find this definition a little misleading, since the general definition of zero-shot is vastly different. With this definition all the supervised models are zero shot models.

> I disagree that there is any data leakage, given the training and testing sets are disjoint.

Can you please kindly point me where the test split is being dropped in your current pipeline when` args.task_name == "pre-training"`?


And how is that the reopening of the issue is disabled? @mononitogoswami
